llvm/lib/SandboxIR/SandboxIR.cpp

```diff
@@ -540,6 +540,48 @@ void StoreInst::dump() const {
   dump(dbgs());
   dbgs() << "\n";
 }
+#endif // NDEBUG
+
+Value *ReturnInst::create(Value *RetVal, Instruction *InsertBefore,
+                          Context &Ctx) {
+  llvm::Instruction *BeforeIR = InsertBefore->getTopmostLLVMInstruction();
+  auto &Builder = Ctx.getLLVMIRBuilder();
+  Builder.SetInsertPoint(BeforeIR);
+  llvm::ReturnInst *NewRI;
+  if (RetVal != nullptr)
+    NewRI = Builder.CreateRet(RetVal->Val);
+  else
+    NewRI = Builder.CreateRetVoid();
+  return Ctx.createReturnInst(NewRI);
+}
+
+Value *ReturnInst::create(Value *RetVal, BasicBlock *InsertAtEnd,
+                          Context &Ctx) {
+  auto &Builder = Ctx.getLLVMIRBuilder();
+  Builder.SetInsertPoint(cast<llvm::BasicBlock>(InsertAtEnd->Val));
```

Comment:
The next few lines of code is duplicated across the two create functions.  Do you want to just set the insert point here and call a private function that does the rest?

Reply:
Yes, I can do that.